toolkits/salesforce/arcade_salesforce/models.py

```diff
@@ -0,0 +1,344 @@
+        ]
+
+    async def get_contact_notes(self, contact: dict, limit: Optional[int] = 10) -> dict:
+        notes = await self.get_notes(contact["Id"], limit)
+        if notes:
+            contact["Notes"] = notes
+        return contact
+
+    async def get_account_leads(self, account_id: str, limit: Optional[int] = 10) -> list[dict]:
+        leads = await self._get_related_objects(
+            SalesforceObject.LEAD, SalesforceObject.ACCOUNT, account_id, limit
+        )
+        return [clean_lead_data(lead) for lead in leads]
+
+    async def get_notes(self, parent_id: str, limit: Optional[int] = 10) -> list[dict]:
+        query = build_soql_query(
+            "SELECT Id, Title, Body, OwnerId, CreatedById, CreatedDate "
+            "FROM Note "
+            "WHERE ParentId = '{parent_id}' "
+            "LIMIT {limit}",
+            parent_id=parent_id,
+            limit=limit,
+        )
+        response = await self.get("query", params={"q": query})
+        notes = response["records"]
+        return [clean_note_data(note) for note in notes]
+
+    # TODO: Add support for retrieving Currency, when enabled in the org account.
+    # If not enabled and we try to retrieve it, we get a 400 error.
+    # More inf: https://developer.salesforce.com/docs/atlas.en-us.254.0.object_reference.meta/object_reference/sforce_api_objects_opportunity.htm#i1455437
+    async def get_account_opportunities(
+        self,
+        account_id: str,
+        limit: Optional[int] = 10,
+    ) -> list[dict]:
+        query = build_soql_query(
+            "SELECT Id, Name, Type, StageName, OwnerId, CreatedById, LastModifiedById, "
+            "Description, Amount, Probability, ExpectedRevenue, CloseDate, ContactId "
+            "FROM Opportunity "
+            "WHERE AccountId = '{account_id}' "
+            "LIMIT {limit}",
+            account_id=account_id,
+            limit=limit,
+        )
+        response = await self.get("query", params={"q": query})
+        opportunities = response["records"]
+        return [clean_opportunity_data(opportunity) for opportunity in opportunities]
+
+    async def get_account_tasks(
+        self,
+        account_id: str,
+        limit: Optional[int] = 10,
+    ) -> list[dict]:
+        tasks = await self._get_related_objects(
+            SalesforceObject.TASK, SalesforceObject.ACCOUNT, account_id, limit
+        )
+        return [clean_task_data(task) for task in tasks]
+
+    async def enrich_account(
+        self,
+        account_id: Optional[str] = None,
+        account_data: Optional[dict[str, Any]] = None,
+        limit_per_association: Optional[int] = 10,
+    ) -> dict:
+        """Enrich account dictionary with notes, leads, contacts, etc.
+
+        Must provide either `account_id` or `account_data`.
+        """
+        if (account_id and account_data) or (not account_id and not account_data):
+            raise ValueError("Must provide either `account_id` or `account_data`")
```

Comment:
Remove ambiguity between OR and XOR
```suggestion
            raise ValueError("Must provide exactly one of `account_id` or `account_data`")
```